src/app/services/backend.service.ts

```diff
@@ -19,8 +19,8 @@ export class BackendService {
     return this.http.post('api/contacts', card).toPromise();
   }
 
-  contactId(id, contact) {
-    return this.http.get(`api/contacts/${id}`).toPromise();
+  getContact(id, contact) {
+    return this.http.get(`api/contacts/${id}`, contact).toPromise();
```

Comment:
You shouldn't be sending a data object with a get request. Get requests don't have a body. Therefore you don't need a `contact` parameter for this method.